### DISCUSSION TITLE
Another Option: Drill Down on an Article or Two at a Time

### DISCUSSION THREAD

**Discussion by topherwhatever** [topherwhatever/Desire-NLP#6] [General]
Park, S.-J., Kang, C.-U., & Byun, Y.-C. (2021). Extreme Gradient Boosting for Recommendation System by Transforming Product Classification into Regression Based on Multi-Dimensional Word2Vec. Symmetry, 13(5), 758. https://doi.org/10.3390/sym13050758

Here's an article looking at feature importance. It's not in the way we
re thinking, but I think we'll be limited at the 2vec level without real augmentation.

Abstract:Now that untact services are widespread and worldwide, the number of users visiting online shopping malls has increased. For example, the recommendation systems in Netflix, Amazon, etc., have gained a lot of attention by attracting many users and have made large profit by recommending suitable products to their users. In the paper, we conduct a study to enhance recommendation accuracy using Word2Vec, widely used in natural language processing. We collect user shopping history with personal click preference information of product items as data, representing a document for natural language processing. The sequence of product item clicks is fed into the Word2Vec technology algorithm to obtain the vectors symmetrically representing all of the product items clicked by users. Training and test data have a series of vectors representing a sequence of the clicked product items as inputs and a purchased product as a target. Machine learning models recommend a product as a symmetric vector for each input and calculate the similarity among the recommended vectors and all other registered products they sell in the system to recommend multiple products as final recommendation results. We use XGBoost regressor and classifier models to recommend some products that users would like and evaluate the recommendation accuracy. A finally recommended product by the models is a vector, and the system recommends some more products by calculating the similarity as mentioned above. We evaluated the classifier model’s recommendation accuracy without Word2Vec encoding first and then with the Word2Vec technique. Meanwhile, we can represent the products with single or multiple dimensional vectors. We noted that the recommendation accuracy increases when we use multiple dimensions of Word2Vec vectors from the experiments. We also evaluated the performances when the system recommends one or multiple products. For the recommendation of multiple products (five here), a regression model has higher accuracy than a classification model in all dimensions of vectors.
